File: 5 - filas/java/Aluno.java
public class Aluno {
    public int matricula;
    public String nome;

    public Aluno(int matricula, String nome) {
        this.matricula = matricula;
        this.nome = nome;
    }

    @Override
    public String toString() {
        return "Aluno [matricula=" + matricula + ", nome=" + nome + "]";
    }
  

    @Override
    public boolean equals(Object obj) {
        
        Aluno other = (Aluno) obj;
        if (matricula != other.matricula)
            return false;
        if (nome == null) {
            if (other.nome != null)
                return false;
        } else if (!nome.equals(other.nome))
            return false;
        return true;
    }

    
}

File: 5 - filas/python/aluno.py
class Aluno:
    
    def __init__(self, matricula, nome):
        self.matricula = matricula
        self.nome = nome
    
    def __str__(self):
        return "Aluno [matricula=" + str(self.matricula) + ", nome=" + self.nome + "]"

    def __eq__(self, obj):
        if (self.matricula != obj.matricula):
            return False
        if (self.nome != obj.nome):
            return False
        return True
    